kernel/exfat-nofuse/Makefile

```diff
@@ -9,15 +9,13 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=exfat-nofuse
-PKG_VERSION=2017-01-03-$(PKG_SOURCE_VERSION)
+PKG_VERSION=snapshot-20170619
 PKG_RELEASE:=1
 
-PKG_SOURCE=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_MIRROR_HASH:=80abb670a59dfa62413e600cee7d13fc65b9980e00579edaa2573fab8568fe93
-PKG_SOURCE_URL:=https://github.com/dorimanx/exfat-nofuse.git
-PKG_SOURCE_PROTO:=git
-PKG_SOURCE_SUBDIR=$(PKG_NAME)-$(PKG_VERSION)
-PKG_SOURCE_VERSION:=8d291f525ce6d88fe0d8b11b86fd5c2e900401d3
+PKG_SOURCE_URL:=https://github.com/dorimanx/exfat-nofuse/tarball/de4c760bc9a05ead83bc3ec6eec6cf1fb106f523?dummy=/
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_HASH:=22f6e9360a3592239efd328febe1acd71b6033c4cd5b8579cadf5b889e46aafc
+PKG_BUILD_DIR:=$(BUILD_DIR)/dorimanx-$(PKG_NAME)-de4c760
```

Comment:
I would suggest not using github auto-genereated tarballs.  Last time I checked there was still no service guarantee that it will provide tarballs with the same hash value

Reply:
It has been working for years for others....
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=194898
https://github.com/freebsd/freebsd-ports/blob/master/Mk/bsd.sites.mk#L345

It's the same as long as you use the same URL. However looking a bit closer it seems that what FreeBSD does makes at bit neater and removes one HTTP redirect.

That would be it a bit better I guess...

```
PKG_SOURCE_URL:=https://codeload.github.com/dorimanx/$(PKG_NAME)/tar.gz/de4c760?dummy=/
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_HASH:=8cf4c0f6b68ca46ac1581fa56ba2b78246fbe7230190ea38c80e46f48fbcd6cd
PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-de4c760
```

Reply:
No.  The first thing is that others already doing it for ages does not justify the action itself.  It may seem harmless and works for now, but it can be still flawed.  People may accept such compromise, but I do not think it not necessary in this case.

About the github archive service, I consulted github people before and they tell me there is no such guarantee as consistent checksum across different downloads.  I still cannot found such service agreement in github api page.

Search and read the bugzilla comments with keyboard "sum" or "cheksum" or "csum", you can find that the csum change thing really happened before and it's possible that it will happen in the future.  If I read it correctly, FreeBSD-ports uses codeload not because it's faster for eliminating a redirect, but because multiple backends may return tarballs with different checksums so they just pinpoint to the codeload backend.

Reply:
As I mentioned, depending on URL it changes (hence the new checksum) however it's been static for years if you use the same URL and used in the FreeBSD repo for quite some time as you can tell no to mention it has a lot more packages using this. Feel free to grep the ports free and you'll see packages that hasn't been updated in 1y+ which still works just fine using this method.  You can have a look at http://beefy9.nyi.freebsd.org/ if you want to verify this, "Failed ports" --> Checksum would be where such issues would show up. There are actually two errors (as of writing) and both  seems to be due to broken recent update of ports. Having a quick look at my about 2w old ports tree 2432 uses this mechanism so I think we can safely assume that it works IRL. :-)